apps/extension/server/src/types.ts

```diff
@@ -10,4 +10,5 @@ export interface ExtractionResult {
 export interface CachedDocument {
   documentText: string;
   extractionResults: ExtractionResult[];
+  undoStack: Array<{ deleted: any[] } | undefined>;
```

Comment:
Yeah, it's on purpose here because I would have to cast it down the road anyway. The code here doesn't really care about those values. They are just kept here so we can pass them back to the `restore` call. `recast` types are really annoying to work with and I wasn't able to find a type that would represent "any" AST node. From what I understand, the problem is that `recast` handles more than a single AST formats and they are not always compatible